packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -138,3 +151,77 @@ class ProcessLister {
     throw StateError('Failed to find $powershell or $pwsh on PATH');
   }
 }
+
+class WindowsVersionExtractor {
+  WindowsVersionExtractor(this.processManager);
+
+  final ProcessManager processManager;
+
+  Future<WindowsVersionExtractionResult> getDetails() async {
+    String? caption, releaseId, displayVersion;
+    try {
+      final ProcessResult captionResult = await processManager.run(
+        <String>['wmic', 'os', 'get', 'Caption,OSArchitecture'],
+      );
+
+      if (captionResult.exitCode == 0) {
+        final String? output = captionResult.stdout as String?;
+        if (output != null) {
+          final List<String> parts = output.split('\n');
+          if (parts.length >= 2) {
+            caption = parts[1].replaceAll('Microsoft Windows', '').replaceAll('  ', ' ').trim();
+          }
+        }
+      }
+    } on ProcessException {
+      // Ignored, use default null value.
```

Comment:
We may want to log the error here with `printTrace` just in case someone reports a bug in this feature and does not want to debug it themselves.